HACKING

```diff
@@ -0,0 +1,20 @@
+ MODIFYING OPENSSL SOURCE
+ ------------------------
+ This document describes the way to add custom modifications to OpenSSL sources.
+ 
+ If you are adding new public functions to the custom library build, you need to
+ either add a prototype in one of the existing OpenSSL header files;
+ or provide a new header file and edit Configurations/unix-Makefile.tmpl to pick up that file.
+ 
+ After that performthe following steps:
+ 
+ ./configure [your-options]
+ make update
+ make
+ 
+ “make update” ensures that your functions declarations are added to util/libcrypto.num or util/libssl.num
+ If you plan to submit the changes you made to OpenSSL (see CONTRIBUTING), it's worth running 
```

Comment:
Maybe it also needs to mention:-
make update also generates files related to errors.
If a merge error occurs in one of these generated files then the generated files need to be removed
and regenerated using make update. To aid in this process the generated files can be committed seperately so they can be removed easily.